plugins/inputs/pgbouncer/pgbouncer.go

```diff
@@ -132,52 +209,84 @@ func (p *PgBouncer) Gather(acc telegraf.Accumulator) error {
 	return poolRows.Err()
 }
 
-type scanner interface {
-	Scan(dest ...interface{}) error
-}
-
-func (p *PgBouncer) accRow(row scanner, columns []string) (map[string]string,
-	map[string]*interface{}, error) {
-	var dbname bytes.Buffer
+func (p *PgBouncer) showLists(acc telegraf.Accumulator) error {
+	// LISTS
+	rows, err := p.DB.Query(`SHOW LISTS`)
+	if err != nil {
+		return err
+	}
 
-	// this is where we'll store the column name with its *interface{}
-	columnMap := make(map[string]*interface{})
+	defer rows.Close()
 
-	for _, column := range columns {
-		columnMap[column] = new(interface{})
+	// grab the column information from the result
+	columns, err := rows.Columns()
+	if err != nil {
+		return err
 	}
 
-	columnVars := make([]interface{}, 0, len(columnMap))
-	// populate the array of interface{} with the pointers in the right order
-	for i := 0; i < len(columnMap); i++ {
-		columnVars = append(columnVars, columnMap[columns[i]])
+	fields := make(map[string]interface{})
+	tags := make(map[string]string)
+	for rows.Next() {
+		tag, columnMap, err := p.accRow(rows, columns)
+		if err != nil {
+			return err
+		}
+
+		if (*columnMap["list"]).(string) != "dns_pending" {
+			fields[(*columnMap["list"]).(string)] = (*columnMap["items"]).(int64)
```

Comment:
Please check your type assertions! We want to make sure that Telegraf does not panic due to some assumptions made on external data...